float64 ang_left
float64 ang_right